tests/test_nblist.py

```diff
@@ -362,3 +362,21 @@ def test_neighborlist_on_subset_of_system():
         assert len(reference_complete_ixns) == len(test_ixn_list), "Number of blocks with interactions don't agree"
 
         assert_ixn_lists_are_equal(reference_complete_ixns, test_ixn_list)
+
+
+@pytest.mark.parametrize("block_size", [32])
+@pytest.mark.parametrize("tiles", [2, 10, 100])
+def test_nblist_max_interactions(block_size, tiles):
+    """Verify that if all particles in the system interact, that the neighborlist correctly assigns large enough buffers"""
+    rng = np.random.default_rng(2023)
+
+    coords = rng.random(size=(block_size * tiles, 3))
```

Comment:
nit: should this be shrunk slightly, to ensure that max(pdist(coords)) < cutoff of 1.2?

(If generated from unit cube, then max distance is like 1.7 (for opposite corners (0,0,0), (1,1,1).)

EDIT: ahh, sorry, the cutoff is defined as 10.0 a few lines later -- for ease of skimming, might name cutoff = 10.0 a few lines earlier, but not necessary -- this is correct and readable as-is

Reply:
Declared explicitly in https://github.com/proteneer/timemachine/pull/1163/commits/acdfdcc3732d530a5481228ddb10b9c482d5de44